internal/exec/stages/files/filesystemEntries.go

```diff
@@ -140,20 +140,34 @@ func (s *stage) createResultFile() error {
 	}
 
 	s.Logger.PushPrefix("createResultFile")
+	var prevReport interface{}
+	prevPath, _ := s.JoinPath(distro.ResultFilePath())
+
+	prevData, err := ioutil.ReadFile(prevPath)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("reading previous report: %w", err)
+	}
+	if err == nil {
+		err = json.Unmarshal(prevData, &prevReport)
+		if err != nil {
+			return fmt.Errorf("couldn't unmarshal previous report output: %v", err)
+		}
+	}
 	defer s.Logger.PopPrefix()
```

Comment:
This needs to be immediately after the `PushPrefix`.  The new error returns are currently failing to pop.